lib/index.d.ts

```diff
@@ -83,11 +83,19 @@ declare namespace Realm {
         inMemory?: boolean;
         schema?: (ObjectClass | ObjectSchema)[];
         schemaVersion?: number;
-        sync?: Realm.Sync.SyncConfiguration;
+        sync?: Partial<Realm.Sync.SyncConfiguration>;
```

Comment:
When you have `PartialConfiguration` 👇 I don't see the need for this to be `Partial` too?